apps/user/serializers.py

```diff
@@ -272,3 +278,169 @@ def save(self, **kwargs):
                 client_ip=client_ip,
                 device_type=device_type)
         )
+
+# ----------------------- NEW GRAPHQL SCHEME Serializers ----------------------------------
+
+
+class LoginSerializer(serializers.Serializer):
+    email = serializers.CharField()
+    password = serializers.CharField(write_only=True)
+    captcha = serializers.CharField(write_only=True, required=False)
+
+    @classmethod
+    def is_captcha_required(cls, user=None, email=None):
+        _user = user or User.objects.filter(email=email).first()
+        return (
+            _user is not None and
+            _user.profile.login_attempts >= settings.MAX_LOGIN_ATTEMPTS_FOR_CAPTCHA
+        )
+
+    def validate_password(self, password):
+        # this will now only handle max-length in the login
+        CustomMaximumLengthValidator().validate(password=password)
```

Comment:
This is a custom validator defined in the serializer. Auth validators are used by Django (this is used by DRF).

Not sure why this is not in the register. Do you remember @k9845?

Reply:
Because many were facing weird validation errors on login for those whose password was created before this implementation. To fix this, we need to only validate max length on login and all others on the register.

Reply:
Okay, thanks.
To summarize, since we don't use Django auth validator for signup before, forcing them in login will make them impossible to log in. So, we only included max-length-validator in login.